### ISSUE TITLE
dns rbl check

### ISSUE THREAD

**Issue by scuben** [matteocorti/check_rbl#20]
It would be great if the tool could also check for dns rbl like dbl.spamhaus.org.

this is a (debug) check against "facebook.com":
```
check_rbl -t 60 -H facebook.com --server=dbl.spamhaus.org -vvv -dddd
[extra-opts] check_rbl -t 60 -H facebook.com --server=dbl.spamhaus.org -vvv -dddd
[DBG] validate(facebook.com, facebook.com)
[DBG] reading...facebook.com
[DBG] name = facebook.com, outstanding = 1
[DBG] reading...EOF.
[DBG] waiting for replies
[DBG] replies received: 1
[DBG] handling a reply
[DBG] Processing answer
[DBG] host = 31.13.92.36
[DBG] callback( facebook.com, 31.13.92.36 )
[DBG] outstanding = 0, eof = 1
Using 60 as global script timeout
Checking facebook.com (31.13.92.36) on 1 server(s)
[DBG] reading...36.92.13.31.dbl.spamhaus.org
[DBG] name = 36.92.13.31.dbl.spamhaus.org, outstanding = 1
[DBG] reading...EOF.
[DBG] waiting for replies
[DBG] replies received: 1
[DBG] handling a reply
[DBG] Processing answer
[DBG] host = 127.0.1.255
[DBG] callback( 36.92.13.31.dbl.spamhaus.org, 127.0.1.255 )
[DBG] callback( 36.92.13.31.dbl.spamhaus.org, 127.0.1.255 )
listed in dbl.spamhaus.org as 127.0.1.255
[DBG] outstanding = 0, eof = 1
CHECK_RBL CRITICAL - facebook.com BLACKLISTED on 1 server of 1 (dbl.spamhaus.org) | servers=1;0;0 time=1s;;
```

one can clearlysee that it checks the ip agains dbl.spamhaus.org receiving a positive response but only because spamhaus will tell you that you queries a dns rbl for an ip.

**Comment by matteocorti** [comment]
See: https://www.spamhaus.org/faq/section/Spamhaus%20DBL

I suppose we would need a different option (e.g., --server-dbl) to tell the plugin if it needs to send the domain or the IP

**Comment by oliverrahner** [comment]
Thanks for the plugin!
I just found it and need the DNS BL feature, so added it.
Only found this issue afterwards :)

My implementation indeed introduces the new option -U for URL. Either -H _or_ -U have to be set.

**Comment by matteocorti** [comment]
Thanks! I just released version 1.5.0

**Comment by oliverrahner** [comment]
For demonstration:

*host (legacy) variant:*
```
 $ ./check_rbl -d -v -H alwaysbutter.icu -s multi.uribl.com
[DBG] validate(alwaysbutter.icu, alwaysbutter.icu)
[DBG] reading...alwaysbutter.icu
[DBG] name = alwaysbutter.icu, outstanding = 1
[DBG] reading...EOF.
[DBG] waiting for replies
[DBG] replies received: 1
[DBG] handling a reply
[DBG] Processing answer
[DBG] host = 80.249.161.131
[DBG] callback( alwaysbutter.icu, 80.249.161.131 )
[DBG] outstanding = 0, eof = 1
Using 15 as global script timeout
Checking 131.161.249.80 on 1 server(s)
[DBG] reading...131.161.249.80.multi.uribl.com
[DBG] name = 131.161.249.80.multi.uribl.com, outstanding = 1
[DBG] reading...EOF.
[DBG] waiting for replies
[DBG] replies received: 1
[DBG] handling a reply
[DBG] callback( 131.161.249.80.multi.uribl.com, <undefined> )
[DBG] callback( 131.161.249.80.multi.uribl.com, <undefined> )
not listed in multi.uribl.com
[DBG] outstanding = 0, eof = 1
CHECK_RBL OK - alwaysbutter.icu BLACKLISTED on 0 servers of 1 | servers=0;0;0 time=0s;;
```

* URL (new) variant:*
```
$ ./check_rbl -d -v -U alwaysbutter.icu -s multi.uribl.com
Using 15 as global script timeout
Checking alwaysbutter.icu on 1 server(s)
[DBG] reading...alwaysbutter.icu.multi.uribl.com
[DBG] name = alwaysbutter.icu.multi.uribl.com, outstanding = 1
[DBG] reading...EOF.
[DBG] waiting for replies
[DBG] replies received: 1
[DBG] handling a reply
[DBG] Processing answer
[DBG] host = 127.0.0.2
[DBG] callback( alwaysbutter.icu.multi.uribl.com, 127.0.0.2 )
[DBG] callback( alwaysbutter.icu.multi.uribl.com, 127.0.0.2 )
listed in alwaysbutter.icu.multi.uribl.com as 127.0.0.2
[DBG] outstanding = 0, eof = 1
CHECK_RBL CRITICAL - alwaysbutter.icu BLACKLISTED on 1 server of 1 (alwaysbutter.icu.multi.uribl.com (127.0.0.2)) | servers=1;0;0 time=0s;;
```
